### PR TITLE
[WIP] Integrations

### PR DESCRIPTION
**Pull request by JPonte**
#### APK
[Download build #10140](http://192.168.10.18:8080/job/Pull%20Request%20Builder/10140/artifact/build/artifact/wire-dev-PR1381-10140.apk)
[Download build #10143](http://192.168.10.18:8080/job/Pull%20Request%20Builder/10143/artifact/build/artifact/wire-dev-PR1381-10143.apk)
[Download build #10144](http://192.168.10.18:8080/job/Pull%20Request%20Builder/10144/artifact/build/artifact/wire-dev-PR1381-10144.apk)
[Download build #10145](http://192.168.10.18:8080/job/Pull%20Request%20Builder/10145/artifact/build/artifact/wire-dev-PR1381-10145.apk)
[Download build #10150](http://192.168.10.18:8080/job/Pull%20Request%20Builder/10150/artifact/build/artifact/wire-dev-PR1381-10150.apk)
[Download build #10151](http://192.168.10.18:8080/job/Pull%20Request%20Builder/10151/artifact/build/artifact/wire-dev-PR1381-10151.apk)
[Download build #10153](http://192.168.10.18:8080/job/Pull%20Request%20Builder/10153/artifact/build/artifact/wire-dev-PR1381-10153.apk)
[Download build #10154](http://192.168.10.18:8080/job/Pull%20Request%20Builder/10154/artifact/build/artifact/wire-dev-PR1381-10154.apk)
[Download build #10155](http://192.168.10.18:8080/job/Pull%20Request%20Builder/10155/artifact/build/artifact/wire-dev-PR1381-10155.apk)
[Download build #10160](http://192.168.10.18:8080/job/Pull%20Request%20Builder/10160/artifact/build/artifact/wire-dev-PR1381-10160.apk)
[Download build #10163](http://192.168.10.18:8080/job/Pull%20Request%20Builder/10163/artifact/build/artifact/wire-dev-PR1381-10163.apk)
[Download build #10165](http://192.168.10.18:8080/job/Pull%20Request%20Builder/10165/artifact/build/artifact/wire-dev-PR1381-10165.apk)
[Download build #10166](http://192.168.10.18:8080/job/Pull%20Request%20Builder/10166/artifact/build/artifact/wire-dev-PR1381-10166.apk)
[Download build #10167](http://192.168.10.18:8080/job/Pull%20Request%20Builder/10167/artifact/build/artifact/wire-dev-PR1381-10167.apk)
[Download build #10168](http://192.168.10.18:8080/job/Pull%20Request%20Builder/10168/artifact/build/artifact/wire-dev-PR1381-10168.apk)
[Download build #10175](http://192.168.10.18:8080/job/Pull%20Request%20Builder/10175/artifact/build/artifact/wire-dev-PR1381-10175.apk)

### CODE REVIEW

**Comment by someoneEsle** [comment]
- [ ] Bug 1
I should not see bots when I'm on the people tab on search UI.

- [ ] Bug 2
Add bot to a conversation. Go to participants view, tap the bot. Its details are not shown. (it's unclear by the checklist you gave me whether this is a bug or something that was not implemented yet)

- [ ] Bug 3
Hint text in the Search UI should be "Search", not "search by name or username"

- [ ] Bug 4
If I select some people in the search UI and then switch tab to services, the selection should be cleared (but the query should not)

**Comment by someoneEsle** [comment]
- [ ] Bug 5
I should not have the buttons to call or video call a bot

**Comment by someoneEsle** [comment]
- [ ] Bug 6
The profile pictures of the bot do not load when going to StartUI>Services>bot

**Comment by someoneEsle** [comment]
- [ ] Bug 7
SearchUI>Services>pick a bot>Add integration
At this point I see a list of conversations, but the vast majority of the conversations I have are not there. I have a feeling we're only displaying group conversations here.

**Comment by someoneEsle** [comment]
- [ ] Bug 8
I cannot add any bot to any of the conversations I have from the conversation details page. I get "invalid conversation" error.

**Comment by someoneEsle** [comment]
build again

**Comment by someoneEsle** [comment]
- [ ] Bug 9
open search UI, services, select a bot, tap create a new conversation. Nothing happens

**Comment by makingthematrix** [comment]
- [ ] Bug 10
1. Open a group conversation with a connected user.
2. Go to that user's participant's view.
3. Click on the "speech bubble" icon in the  bottom-left corner to open a private conversation.
4. "back press".
A blank screen appears and you can't go back. Clicking "back" again closes the app.

File: app/src/main/scala/com/waz/zclient/integrations/IntegrationDetailsFragment.scala [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+        if (integrationDetailsController.addingToConversation.nonEmpty) {
+          inject[INavigationController].setRightPage(Page.PICK_USER, IntegrationDetailsFragment.Tag)
+        } else {
+          inject[INavigationController].setLeftPage(Page.PICK_USER, IntegrationDetailsFragment.Tag)
+        }
+    }
+    true
+  }
+
+  def close(): Boolean = {
```
[the remark this answers is not in the document]
Comment (makingthematrix) [reply app/src/main/scala/com/waz/zclient/integrations/IntegrationDetailsFragment.scala]: This one was fixed by Joao.

For the other comments and the `ParticipantBodyFragment` rewriting I created a new PR: https://github.com/wireapp/wire-android/pull/1394
`ParticipantBodyFragment` was changed quite a lot so I thought this way we could avoid some merging problems.

**Comment by someoneEsle** [comment]
- [ ] Bug 10
When I search for a bot in the personal tab, I should see the group chats that the bot is part of

**Comment by someoneEsle** [comment]
- [ ] Bug 10
Searching for a group that has a bot in it doesn't always work

- [ ] Bug 11
guest label in participants view is not displayed properly

**Comment by someoneEsle** [comment]
- [ ] Bug 12
User B is a team user from team Tb, and is guest in a group conversation G created by A, another team user from team Ta. When B opens Search UI>Services>selects a bot>Add service>selects conversation G, nothing happens. Now, we know that guests shouldn't be able to add bots to a conversation, so I'm guessing that we are silently failing to add the bot to conversation G, but the user doesn't know this and there is no warning message about it.

**Comment by someoneEsle** [comment]
- [ ] Bug 13
Guests can remove bots from my conversations

**Comment by someoneEsle** [comment]
- [ ] Problem 13
I can create an infinite amount of 1:1 conversations with the same bot. SearchUI>select bot>create new conversation always creates a new conversation, even if I already have a 1:1 with the bot

**Comment by someoneEsle** [comment]
- [ ] Bug 15
I cannot mute conversation with bot

**Comment by someoneEsle** [comment]
- [ ] Bug 16
Adding a bot to a verified conversation should degrade it

**Comment by someoneEsle** [comment]
- [ ] Bug 17
please bring back the call/video call options ^^

**Comment by someoneEsle** [comment]
- [ ] @Dean Bug 18
Long tap a group conversation. Tap delete. Check "leave" box, confirm delete. Now go to the archive. Select the same conversation. Tap delete again. You are now prompted to leave the conversation (which you already did).
[This is a bug that is currently also in internal, so not new, but I am reporting it because I thought the purpose of you rewriting the menu was to fix these bugs]

**Comment by someoneEsle** [comment]
- [ ] Check tracking

**Comment by someoneEsle** [comment]
- [ ] Bug 20
bots have no profile pictures in the Search UI after fresh install/first login

**Comment by someoneEsle** [comment]
- [ ] Bug 21
Create a 1:1 with a bot. Tap top toolbar. Add people. Add guest. Result: weird warning and then the conversation turns into a conversation with 0 participants

**Comment by olivereitalu** [comment]
Design review: 

— Tab indicator line should be 1px
— No matches found message is missing
— The service icons don’t show up in the list/results and are circles not squares
— All bot avatars should have a white square background (with rounded corners), currently Don has a rectangular avatar with transparency
— Bot icons inside the conversation also need to be square
— Can’t find the conversation I want to add the bot to (group with 3 people)
— Bot avatars are missing a 0.08 alpha border around the avatar (on all screens)
— Removing bots not implemented yet?
— The BOT indicator next to the bot name in the conversation is slightly lower than the name and should be light not bold
— Can’t see Git messages on Android (probably not client related)

**Comment by someoneEsle** [comment]
- [ ] Question 22
Search UI> services> add bot. Shouldn't we sort the conversations by most recently used? Alphabetical order make little sense imho

**Comment by someoneEsle** [comment]
- [ ] Bug 22
I don't see the "conversation is now verified" message in the conversation view after verifying all the devices

**Pull request merged**